src/index.ts

```diff
@@ -3,6 +3,7 @@ import {RlpDecode, RlpEncode, RlpItem, RlpList} from 'rlp-stream';
 const originalNode = require('./trieNode');
 const matchingNibbleLength = require('./util').matchingNibbleLength;
 const keccak = require('keccak');
+const patriciaTree = require('patricia-tree');
```

Comment:
What is the purpose of the patricia tree then? It sounds like you just need a map (hash -> node)

Reply:
What I am doing here is that for the keys that are being read, I'm constructing a tree which is a  concise representation of all the keys eliminating the duplicate prefixes. Let us call this keyTree. For example,
keyStrings: abcd, abxy
    KeyTree: (ab) -> [(cd), (xy)]
(Merkle patricia tree and patricia tree both of them can be used for this purpose.) 

Then I perform a DFS on the KeyTree and read the corresponding values from the merkle-patricia-tree by performing a single traversal on the merkle-patricia-tree.

As we don't need a tree that computes hashes when we read from the tree, I used a basic patricia-tree instead of a merkle-patricia-tree. We might end up re-designing the patricia-tree if we are going to use maps for this purpose.

Reply:
I see now. Okay, that makes more sense. 

I don't think you need a tree, though. You really just need to convert each key into a nibble array, and then radix sort the resulting array. Then, you could implement DFS easily just by running on each key in the array.

Reply:
I understand your solution. I could use radix sort on the array of keys. Basically what I'm doing here is very similar - I'm using a DFS traversal on patricia-tree to perform radix sort. I don't completely understand the advantages of using a sorting algorithm instead of a patricia-tree.

If it is about the conversion of buffer to a string, I can have a function similar to the function that converts buffers to an array of nibbles, to convert buffer to a string, instead of using the costly Buffer.toString function.

Am I probably missing out on something here?

Reply:
There are two overheads I'm concerned with: first is the memory allocation overhead of building the tree and second, the benefits of in place traversal. 

But perhaps this is all moot if you can show a significant performance improvement already? What is the improvement?

Reply:
Also, a implementation using a sorted array seems simpler?

Reply:
For 100 random keys of length 7, querying all the keys using get vs getBulk,
averaged over 10 iterations:
getTime: 51.5527ms
getBulkTime: 9.19ms

Reply:
However, I see your point. I'll look into using radix-sort over an array of keys and performing DFS by stepping through the sorted array of keys. It should be simple.